include/seqan3/core/doxygen_only.hpp

```diff
@@ -0,0 +1,91 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file core/doxygen_only.hpp
+ * \defgroup core
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Contains "global" documentation that doesn't fit anywhere else.
+ */
+
+#pragma once
+
+/*!
+ * \namespace seqan3
+ * \brief The main SeqAn3 namespace.
+ */
+/*!
+ * \namespace seqan3::view
+ * \brief The SeqAn3 namespace for views.
+ *
+ * SeqAn3 makes heavy use of views as defined in the
+ * [Ranges Technical Specification](http://en.cppreference.com/w/cpp/experimental/ranges). Currently the
+ * implementation is based on the [range-v3 library](https://github.com/ericniebler/range-v3).
+ *
+ * Since views often have name clashes with regular functions they are implemented in the sub namespace `view`.
+ *
+ * \sa http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/n4651.pdf
+ */
+/*!
+ * \namespace seqan3::literal
+ * \brief The SeqAn3 namespace for literals.
+ *
+ * SeqAn implements "user defined" literals in multiple places, e.g. `auto foo = "ACGTG"_dna4`. These
+ * make working with small examples and tests a lot easier, but the risk of having a name collision with
+ * another library is higher so follow the example of the standard library and define all our literals
+ * in the namespace `seqan3::literal`.
+ *
+ * \attention
+ * This means you cannot use them, unless you explicitly add `using namespace seqan3::literal;` (in addition
+ * to `using namespace seqan3;`).
+ */
+/*!\cond DEV
+ * \namespace seqan3::detail
+ * \brief The internal SeqAn3 namespace.
+ * \details
+ * The contents of this namespace are not visibile to consumers of the library and the documentation is
+ * only generated for developers.
+ * \sa https://github.com/seqan/seqan3/wiki/Documentation
+ * \endcond
+ */
+
+/*!\mainpage SeqAn3 API documentation
+ *
+ * This is the API documentation for the SeqAn3 library.
+ *
+ * We recommend browsing the documentation via the "Modules" menu on the left.
+ *
+ * Installation instructions are available on [GitHub](https://github.com/seqan/seqan3).
+ *
+ * A manual will be added in the future.
+ */
```

Comment:
Maybe we can add a general landing page, that can be added to the doxygen generator?
Or maybe put it into seqan3/core.hpp?